modules/ppcp-button/src/Helper/ThreeDSecure.php

```diff
@@ -118,42 +134,40 @@ public function return_decision( int $decision, Order $order ) {
 	 *
 	 * @return int
 	 */
-	private function no_liability_shift( AuthResult $result ): int {
+	private function no_liability_shift( AuthResult $result ) : int {
+		$enrollment     = $result->enrollment_status();
+		$authentication = $result->authentication_result();
 
-		if (
-			$result->enrollment_status() === AuthResult::ENROLLMENT_STATUS_BYPASS
-			&& ! $result->authentication_result()
-		) {
-			return self::PROCEED;
-		}
-		if (
-			$result->enrollment_status() === AuthResult::ENROLLMENT_STATUS_UNAVAILABLE
-			&& ! $result->authentication_result()
-		) {
-			return self::PROCEED;
-		}
-		if (
-			$result->enrollment_status() === AuthResult::ENROLLMENT_STATUS_NO
-			&& ! $result->authentication_result()
-		) {
-			return self::PROCEED;
+		if ( ! $authentication ) {
+			if ( $enrollment === AuthResult::ENROLLMENT_STATUS_BYPASS ) {
+				return self::PROCEED;
+			}
+
+			if ( $enrollment === AuthResult::ENROLLMENT_STATUS_UNAVAILABLE ) {
+				return self::PROCEED;
+			}
+
+			if ( $enrollment === AuthResult::ENROLLMENT_STATUS_NO ) {
+				return self::PROCEED;
+			}
 		}
```

Comment:
This can be also simplified.

```suggestion
		if ( ! $authentication && in_array( $enrollment, [
			AuthResult::ENROLLMENT_STATUS_BYPASS,
			AuthResult::ENROLLMENT_STATUS_UNAVAILABLE,
			AuthResult::ENROLLMENT_STATUS_NO,
		], true ) ) {
			return self::PROCEED;
		}
```